src/lsm/segmented_array.zig

```diff
@@ -33,10 +34,17 @@ pub fn SortedSegmentedArray(
     comptime Key: type,
     comptime key_from_value: fn (*const T) callconv(.Inline) Key,
     comptime compare_keys: fn (Key, Key) callconv(.Inline) math.Order,
+    comptime options: Options,
```

Comment:
Could we switch to a single `SegmentedArray` constructor and move `Key`/`key_from_value`/`compare_keys` into `options.sorted: ?struct { Key, key_from_value, compare_keys }`? Since we have an `Options` anyway may as well use it :smile:.

Reply:
Potentially, but the current type is:

```
    comptime Key: ?type,
    comptime key_from_value: if (Key) |K| (fn (*const T) callconv(.Inline) K) else void,
    comptime compare_keys: if (Key) |K| (fn (K, K) callconv(.Inline) math.Order) else void,
```

We can't put that type in a struct. Maybe:

```
const Config = struct {
    Key: ?type,
    key_from_value: ?(fn (*const T) callconv(.Inline) K),
    compare_keys: ?(fn (K, K) callconv(.Inline) math.Order),
    verify: bool,
};
```

Reply:
Ah, the problem is that `key_from_value` and `compare_keys` need access to `Key`. That works for arguments, but not when they are fields on a struct. :thinking:

Reply:
What is `Options` was itself `fn Options(Key: ?type)`. Then we continue to expose both segmented array constructors, where the sorted one takes `Key: type, options: Options(Key)` as arguments.

Reply:
The unsorted constructor would still need to take Options(null) for the verify arg. We could do some type-hackery to provide default values for the other fields only when Key==null, but it's starting to seem like over-complicated.